cyclonedx-bom/src/models/metadata.rs

```diff
@@ -41,6 +44,15 @@ pub struct Metadata {
 }
 
 impl Metadata {
+    /// Constructs a new `Metadata` with a timestamp based on the current time
+    /// ```
+    /// use cyclonedx_bom::models::metadata::Metadata;
+    ///
+    /// let metadata = Metadata::new();
+    /// ```
+    /// # Errors
+    ///
+    /// Returns an error variant if unable to generate a valid timestamp
```

Comment:
I'm open to suggestions on this description. It's based on the Rust API docs style guide.